rskj-core/src/main/java/org/ethereum/core/SignatureCache.java

```diff
@@ -20,27 +20,35 @@
 package org.ethereum.core;
 
 import co.rsk.core.RskAddress;
+import co.rsk.crypto.Keccak256;
 import co.rsk.remasc.RemascTransaction;
+import co.rsk.util.MaxSizeHashMap;
 
+import java.util.Collections;
 import java.util.Map;
 
 public abstract class SignatureCache {
-    protected Map<Transaction, RskAddress> addressesCache;
 
-    public abstract RskAddress getSender(Transaction transaction);
+    protected Map<Keccak256, RskAddress> addressesCache;
```

Comment:
as an option, `addressesCache` could be marked as `final` - to emphasise that its initialisation should be done in the ctor

Reply:
Done!